arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -3665,4 +3715,34 @@ std::shared_ptr<StorageSnapshot> RocksDBEngine::currentSnapshot() {
   }
 }
 
+void RocksDBEngine::removeEmptyJournalFilesFromArchive() {
+  LOG_TOPIC("50812", DEBUG, Logger::ENGINES)
+      << "scanning WAL archive directory for empty files...";
+
+  std::string archiveDirectory =
+      basics::FileUtils::buildFilename(_dbOptions.wal_dir, "archive");
+
+  WRITE_LOCKER(lock, _walFileLock);
```

Comment:
lazy lock acquiring?

Reply:
Would be possible, but it is not important at this part of the code. The method is only called once during startup, and there are no concurrent readers or writers for `_walFileLock` at this point.
The lock is only here for consistency with other parts of the code.

Reply:
Ok